cmd/sos.go

```diff
@@ -56,6 +71,29 @@ func newSOSClient(certsFile string) (*sosClient, error) {
 		err error
 	)
 
+	if certsFile == "" && runtime.GOOS == "windows" {
+		// Check if the directory of the "exo" executable contains a file named "sos-certs.pem"
+		// to load the certificate chain from. This is done to work around Golang issue #16736
+		// on Windows ( https://github.com/golang/go/issues/16736 )
+		path, err := os.Executable()
+		if err == nil {
+			dir, err := filepath.Abs(filepath.Dir(path))
+			if err == nil {
+				tmpCertsFile := filepath.Join(dir, "sos-certs.pem")
+				fmt.Println(tmpCertsFile)
+				stat, err := os.Stat(tmpCertsFile)
+				if err == nil && stat.IsDir() == false {
+					certsFile = tmpCertsFile
+				} else {
+					log.Warn(
+						"It seems you are running on Windows and your sos-certs.pem file is missing.\n" +
+							"Please download and extract all files from the exo cli release, not just the executable.\n" +
+							"Please see the 'exo sos help' command for more details.")
+				}
+			}
+		}
+	}
+
```

Comment:
Ignoring errors is a dangerous bet that can lead to hours wasted on debugging obscure error messages that our users may report to us in the future (speaking from experience). A more typical way to implement this code is to check for errors early and handle them properly, which also grants a less indented code:

```go
	if err != nil {
		// Handle error
	}
	
	dir, err := filepath.Abs(filepath.Dir(path))
	if err != nil {
		// Handle error
	}

	tmpCertsFile := filepath.Join(dir, "sos-certs.pem")
	stat, err := os.Stat(tmpCertsFile)
	if err != nil {
		log.Warn(
			"It seems you are running on Windows and your sos-certs.pem file is missing.\n" +
			"Please download and extract all files from the exo cli release, not just the executable.\n" +
			"Please see the 'exo sos help' command for more details.")
		// Maybe return with error here instead of continuing
	}

	certsFile = tmpCertsFile
```

Reply:
OK, I overhauled that, but I'm unsure if I'm supposed to use log.Warn as it prints a JSON-formatted log message, which is not very user friendly. Any advice on that @falzm ?

Reply:
You've imported the `log.Warn()` method from a new package `github.com/labstack/gommon`, which seems to log in JSON format by default. I'd prefer you use the standard `fmt.Fprintln(os.Stderr, "your message [...]")` call for this, please.

Reply:
Whoopsies, removed.